### PR TITLE
Added Azure.MariaDB.UseSSL

### LINKED ISSUE #1852: Azure Database for MariaDB servers should only accept encrypted connections
# Rule request

## Suggested rule change

Azure Database for MariaDB servers should only accept encrypted connections.

If `properties.sslEnforcement` is set to disable both unencrypted and encrypted connections are allowed and there is no proper way to differentiate between them. 

Security pillar for this one.

## Applies to the following

The rule applies to the following:

- Resource type: **[Microsoft.DBforMariaDB/servers]**

## Additional context

[SSL/TLS connectivity in Azure Database for MariaDB](https://learn.microsoft.com/azure/mariadb/concepts-ssl-connection-security)
[Template reference](https://learn.microsoft.com/azure/templates/microsoft.dbformariadb/servers)

### PR DESCRIPTION
**Pull request by BenjaminEngeset**
## PR Summary

Fixes #1852 

Added Azure.MariaDB.UseSSL rule.

## PR Checklist

- [x] PR has a meaningful title
- [x] Summarized changes
- [x] Change is not breaking
- [x] This PR is ready to merge and is not **Work in Progress**
- **Rule changes**
  - [x] Unit tests created/ updated
  - [x] Rule documentation created/ updated
  - [x] Link to a filed issue
  - [x] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section
- **Other code changes**
  - [ ] Unit tests created/ updated
  - [ ] Link to a filed issue
  - [ ] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section

### CODE REVIEW

**Comment by bengeset96** [comment]
@BernieWhite 

Should be ready for review. Similar to the MySQL version. Since its default enabled I choose to not include Bicep and ARM examples. Was considering to use `UseTLS`, but since MS documentation is refering to it as SSL it's probably better suited for customers with `UseSSL`.

**Review by BernieWhite — COMMENTED** [review]
@bengeset96 This rule also seem to be included in PR #1860. Do we want to just close this PR and use #1860 to merge both rules.

**Comment by bengeset96** [comment]
> @bengeset96 This rule also seem to be included in PR #1860. Do we want to just close this PR and use #1860 to merge both rules.

Yes, we can do this. What is your branch recommendation/strategy when developing multiple rules?

**Comment by BernieWhite** [comment]
> > @bengeset96 This rule also seem to be included in PR #1860. Do we want to just close this PR and use #1860 to merge both rules.
> 
> Yes, we can do this. What is your branch recommendation/strategy when developing multiple rules? For now I'm creating branches after the previous rule branch.

Preference is one rule / one PR. But a small number of related rules is ok in a single PR. Adding too many rules or unrelated rules just makes it harder to review.

If you want to keep the changes in this PR (preferred) just remove the `Azure.MariaDB.UseSSL` content from the other PR otherwise I'll be reviewing the same content twice.

You changes in the other PR are here: https://github.com/Azure/PSRule.Rules.Azure/pull/1860/files see you they include changes for both rules.

**Comment by bengeset96** [comment]
> > > @bengeset96 This rule also seem to be included in PR #1860. Do we want to just close this PR and use #1860 to merge both rules.
> > 
> > 
> > Yes, we can do this. What is your branch recommendation/strategy when developing multiple rules? For now I'm creating branches after the previous rule branch.
> 
> Preference is one rule / one PR. But a small number of related rules is ok in a single PR. Adding too many rules or unrelated rules just makes it harder to review.
> 
> If you want to keep the changes in this PR (preferred) just remove the `Azure.MariaDB.UseSSL` content from the other PR otherwise I'll be reviewing the same content twice.
> 
> You changes in the other PR are here: https://github.com/Azure/PSRule.Rules.Azure/pull/1860/files see you they include changes for both rules.

Done. Removed `Azure.MariaDB.UseSSL` related in 1860.

**Comment by bengeset96** [comment]
@BernieWhite 

Fixed the merge conflicts, but CI issue. How can I get by this? I've checked the rule documentation, I can't find the failure.

**Comment by bengeset96** [comment]
> > @BernieWhite
> > Fixed the merge conflicts, but CI issue. How can I get by this? I've checked the rule documentation, I can't find the failure.
> 
> Hi @bengeset96 the name of the markdown file has an extra dot in it.

Thanks! Fixed now.

File: docs/en/rules/Azure.MariaDB.UseSSL.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+Azure Database for MariaDB should be configured to only accept encrypted connections.
+Unless explicitly required, consider enabling _enforce SSL connections_.
+
+Also consider using Azure Policy to audit or enforce this configuration.
+
+## LINKS
+
+- [Data encryption in Azure](https://learn.microsoft.com/azure/architecture/framework/security/design-storage-encryption#data-in-transit)
+- [SSL connectivity in Azure Database for MariaDB](https://learn.microsoft.com/azure/mariadb/concepts-ssl-connection-security)
```
Comment (BernieWhite) [inline docs/en/rules/Azure.MariaDB.UseSSL.md:33]: Lets add a link to the Bicep deployment reference.

File: docs/en/rules/Azure.MariaDB.UseSSL.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+Unencrypted communication to MariaDB server instances could allow disclosure of information to an untrusted party.
+
+## RECOMMENDATION
+
+Azure Database for MariaDB should be configured to only accept encrypted connections.
+Unless explicitly required, consider enabling _enforce SSL connections_.
+
+Also consider using Azure Policy to audit or enforce this configuration.
+
```
Comment (BernieWhite) [inline docs/en/rules/Azure.MariaDB.UseSSL.md:29]: We need some code examples for this ARM and Bicep.

**Review by BernieWhite — CHANGES_REQUESTED** [review]
@bengeset96 Rule looks good. Need to add a few things to the docs then good to go.

File: docs/en/rules/Azure.MariaDB.UseSSL.md (hunk 1 above)
Comment (bengeset96) [reply docs/en/rules/Azure.MariaDB.UseSSL.md:33]: I'll do that.

File: docs/en/rules/Azure.MariaDB.UseSSL.md (hunk 2 above)
Comment (bengeset96) [reply docs/en/rules/Azure.MariaDB.UseSSL.md:29]: Yes, I'll fix that. I took another `UseSSL` rule as reference. I'll update that as well later on with some ARM and Bicep examples.

**Comment by bengeset96** [comment]
@BernieWhite 

I've added Bicep and ARM code examples now and added the template reference.

**Review by BernieWhite — APPROVED** [review]
Thanks @BenjaminEngeset. Great work. All good to merge.

**Comment by BenjaminEngeset** [comment]
@microsoft-github-policy-service agree

**Comment by BenjaminEngeset** [comment]
@BernieWhite 

Now CLA is fixed.

**Pull request merged**